setup.py

```diff
@@ -44,7 +44,6 @@ def find_version(*paths):
         "s3fs == 2022.2.0",
         "tenacity == 8.1.0",
         "typing-extensions == 4.4.0",
-        "memray == 1.6.0",
     ],
```

Comment:
It might be nice to also provide this optionally if we are supporting installation through setuptools: https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies

Reply:
Thanks for the link. I'm OK keeping memray in `dev-requirements.txt` for simplicity for now. It might also be a bit weird if we wind up making production deploys of DeltaCAT that sometimes do and sometimes don't have profiling enabled based on the underlying architecture they're deployed to so, either way, I think we should continue to separate it out as a non-prod dependency for now (unless we can optionally include profiling for all supported production architectures, and without any undesirable side-effects).